stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -26,16 +32,45 @@ internal class PaymentSheetViewModel internal constructor(
     private val publishableKey: String,
     private val stripeAccountId: String?,
     private val stripeRepository: StripeRepository,
+    private val stripe: Stripe,
     private val workContext: CoroutineContext = Dispatchers.IO
 ) : AndroidViewModel(application) {
     private val mutableError = MutableLiveData<Throwable>()
     private val mutableTransition = MutableLiveData<TransitionTarget>()
     private val mutablePaymentMethods = MutableLiveData<List<PaymentMethod>>()
     private val mutableSelection = MutableLiveData<PaymentSelection?>()
+    private val mutablePaymentIntentResult = MutableLiveData<PaymentIntentResult>()
     internal val paymentMethods: LiveData<List<PaymentMethod>> = mutablePaymentMethods
     internal val error: LiveData<Throwable> = mutableError
     internal val transition: LiveData<TransitionTarget> = mutableTransition
     internal val selection: LiveData<PaymentSelection?> = mutableSelection
+    internal val paymentIntentResult: LiveData<PaymentIntentResult> = mutablePaymentIntentResult
+
+    internal constructor(
+        application: Application,
+        publishableKey: String,
+        stripeAccountId: String?,
+        stripeRepository: StripeRepository,
+        workContext: CoroutineContext = Dispatchers.IO
+    ) : this(
+        application,
+        publishableKey,
+        stripeAccountId,
+        stripeRepository,
+        Stripe(
```

Comment:
Can we depend on just `StripeRepository` and `PaymentController` instead of `Stripe?

Reply:
Do you mean create a `Stripe` object internally as opposed to passing it in from the constructor? Or do you mean not use a `Stripe` object at all can manually call `paymentController.startConfirmAndAuth`?

Reply:
Yeah, I mean don't create a `Stripe` object. Not sure how much that complicates things.

Reply:
Right. Yeah, I think that's reasonable. `confirmPayment` is just a wrapper around `paymentController.startConfirmAndAuth` so we can just call that directly.